dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -488,6 +488,12 @@ private String getResourceIds(ProcessData processData) {
             putMsg(result, Status.PROCESS_DEFINE_STATE_ONLINE, processDefinitionId);
             return result;
         }
+        // check process instances is already running
+        List<ProcessInstance> processInstances =  processInstanceMapper.queryByProcessDefineIdAndStatus(processDefinitionId, Constants.NOT_TERMINATED_STATES);
+        if (CollectionUtils.isNotEmpty(processInstances)) {
+            putMsg(result, Status.DELETE_PROCESS_DEFINITION_BY_ID_FAIL,processInstances.size());
+            return result;
+        }
```

Comment:
```suggestion
        List<ProcessInstance> processInstances =  processInstanceService.queryByProcessDefineIdAndStatus(processDefinitionId, Constants.NOT_TERMINATED_STATES);
        if (CollectionUtils.isNotEmpty(processInstances)) {
            putMsg(result, Status.DELETE_PROCESS_DEFINITION_BY_ID_FAIL,processInstances.size());
            return result;
        }
```

Hi,
In `processDefinitionService` service layer, it is not recommended straightly use `processInstanceMapper` dao layer, this part should be in `processInstanceService`, use `processInstanceService` here.

Reply:
Processinstanceservice has been used. Processinstanceservice should implement the interface. There are contributors who have submitted branches. I have not changed this.

---

已使用processInstanceService ，processInstanceService 应该实现接口，已经有贡献者提交分支，这块我没有改.